src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6799,7 +6799,7 @@ private bool ContainingMethodOrLambdaRequiresValue()
             return
                 (object)containingMethod == null ||
                     !containingMethod.ReturnsVoid &&
-                    !containingMethod.IsTaskReturningAsync(this.Compilation);
+                    containingMethod.AsyncKind != AsyncKind.Task;
         }
```

Comment:
The logic of this method is

* non-async, void -> false
* non-async, non-void -> true
* async, AsyncKind.NotAsync -> (I trust this is impossible??)
* async, Invalid -> true
* async, Void -> (already caught by containingMethod.ReturnsVoid)
* async, Task -> false
* async, Task_T -> true

I wonder about that AsyncKind.Invalid case. Should it maybe return FALSE from this method `ContainingMethodRequiresValue` ? (I can't think of an example...)